public/app/plugins/panel/geomap/utils/location.ts

```diff
@@ -199,3 +241,16 @@ function getPointsFromGeohash(field: Field<string>): Point[] {
   }
   return points;
 }
+
+function getPointsFromLookup(field: Field<string>, locationData: LocationKey[]): Point[] {
+  const count = field.values.length;
+  const points = new Array<Point>(count);
+  for (let i = 0; i < count; i++) {
+    const target = field.values.get(i);
+    const location = locationData.filter((loc: LocationKey) => loc.key.toUpperCase() === target.toUpperCase())[0];
```

Comment:
This will be a super inefficient way to lookup points :grimacing: 

Rather than loading the initial data into an array -- load it into a `Map<string,Location>`  then the lookups (called frequently!) will be fast and the cost is paid once when you load the lookup file

Reply:
Changed location data to be a Map<string, Location>!